src/position-manager/AllowancePositionManager.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity 0.8.28;
+
+import {SignatureChecker} from 'src/dependencies/openzeppelin/SignatureChecker.sol';
+import {SafeERC20, IERC20} from 'src/dependencies/openzeppelin/SafeERC20.sol';
+import {EIP712} from 'src/dependencies/solady/EIP712.sol';
+import {MathUtils} from 'src/libraries/math/MathUtils.sol';
+import {NoncesKeyed} from 'src/utils/NoncesKeyed.sol';
+import {EIP712Hash, EIP712Types} from 'src/position-manager/libraries/EIP712Hash.sol';
+import {PositionManagerBase} from 'src/position-manager/PositionManagerBase.sol';
+import {ISpokeBase} from 'src/spoke/interfaces/ISpokeBase.sol';
+import {IAllowancePositionManager} from 'src/position-manager/interfaces/IAllowancePositionManager.sol';
+
+/// @title AllowancePositionManager
+/// @author Aave Labs
+/// @notice Position manager to handle withdraw permit, credit delegation and borrow actions on behalf of users.
+contract AllowancePositionManager is
+  IAllowancePositionManager,
+  PositionManagerBase,
+  NoncesKeyed,
+  EIP712
+{
+  using SafeERC20 for IERC20;
+  using EIP712Hash for *;
+  using MathUtils for uint256;
+
+  /// @notice Mapping of withdraw allowances.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _withdrawAllowances;
+
+  /// @notice Mapping of credit delegations.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _creditDelegations;
```

Comment:
i dn't be opposed to use similar terms (e.g. `_borrowAllowances`) for credit delegation in general... But fine to keep it as it is if "giving borrow allowance" sounds "weird" and industry is used to "delegating credit"